### ISSUE TITLE
Reexport Base.imag?

### ISSUE THREAD

**Issue by jiahao** [JuliaGeometry/Quaternions.jl#29]
Currently, `Base.imag` is [not imported and re-exported](https://github.com/JuliaGeometry/Quaternions.jl/blob/master/src/Quaternions.jl#L8), which causes shadowing errors when intermixing `Complex` and `Quaternion` numbers in arithmetic computations. Since `im^2 == -1` in both algebras, the algebras themselves are not in conflict, and the analogous mixed-algebra computations should be similar to `Real * Complex` in `Base`. At least on paper, I don't foresee any problems in allowing `Quaternions.imag` to extend `Base.imag`.

**Comment by JeffBezanson** [comment]
The `imag` function here returns a vector of the 3 imaginary components. Does that really implement the `imag` function? Doesn't seem right to me.

**Comment by chakravala** [comment]
If you'd like to gain a better understanding geometric or algebraic understanding of imaginary units of quaternions and complex values, I recommend you play around with it on [Grasmann.jl](https://gitbub.com/chakravala/Grassmann.jl) a bit.

```Julia

julia> using Grassmann

julia> i,j,k = hyperplanes(ℝ^3)
3-element Array{SValue{⟨+++⟩,2,B,Int64} where B,1}:
 -1v₂₃
 1v₁₃ 
 -1v₁₂

julia> (ℝ^3)(I) # this is the 3D pseudoscalar
1v₁₂₃

julia> (ℝ^3)(I)^2 # squares to -1 also
-1v
```
In the quaternion algebra, there is an `i,j,k` which are all **planes** which square to `-1` but there is another capital `I` which is usually called the **volume** element which is a single imaginary unit.

Quaternions are generated by a 3D vector space, but complex numbers are generated by a 2D vector space combining a scalar value with a pseudoscalar value, i.e the volume element `V(I)`.
```Julia
julia> (ℝ^2)(I) # this is the 2D pseudoscalar
1v₁₂

julia> (ℝ^2)(I)^2 # squares to -1 as imaginary unit
-1v
```
As you can see, the pseudoscalar `V(I)` in 2D is `v12` while in 3D it is `v123`.

It so happens that the pseudoscalar `v12` from 2D is equal to `i` from the `i,j,k` of quaternions also.

Thus, you could either:
- equate `im` with `v12`, the bivector plane element, which would require a sign change for 3D!
- equate `im` with `V(I)`, the pseudoscalar element, which is more natural for `n`-dimensions

If you associate it with the `v12` element alone, then the `Base.imag` function for quaternions should only return the `i` component of the quaternions instead of all 3 values. Otherwise, if you choose to equate `im` with the value of the pseudoscalar volume element `V(I)`, then there is simply no comparison between the `im` value and the `i,j,k` components of the quaternions, they are fundamentally different quantities.

I hope that this explanation provides some insight about the relationship between `im` and `I` and `i`.

Basically, `im` is a 2D [pseudoscalar](https://en.wikipedia.org/wiki/Pseudoscalar), while `i,j,k` is both a 3D [pseudovector](https://en.wikipedia.org/wiki/Pseudovector) and also a [paravector](https://en.wikipedia.org/wiki/Paravector).

**Comment by goretkin** [comment]
The definitions at

https://github.com/JuliaGeometry/Quaternions.jl/blob/01d45b4f345fb91945211d3c6387da4b5934e7d7/src/Quaternion.jl#L15

and

https://github.com/JuliaGeometry/Quaternions.jl/blob/01d45b4f345fb91945211d3c6387da4b5934e7d7/src/Quaternion.jl#L21

would be consistent with defining

`Base.imag(q::Quaternion) = q.v1`

However, while I can personally attest to finding it useful to have conversion/promotion from Real to Complex or Real to Quaternion, I do not have the same experience with regards to mixing Complex and Quaternion numbers in operations. Does anyone have an example where having the preceding definitions were beneficial?

**Comment by hyrodium** [comment]
As discussed in #61, we will have `imag_part` function which returns `Tuple` instead of `Vector`. (#77)
The original `Quaternions.imag` will be deprecated with the PR, so I'll close this issue.
